src/qnetvo/qnodes.py

```diff
@@ -95,3 +95,46 @@ def circuit(settings):
         return qml.probs(wires=network_ansatz.layers_wires[-1])
 
     return circuit
+
+
+def density_matrix_qnode(network_ansatz, **qnode_kwargs):
+    """Constructs a qnode that computes the density matrix in the computational basis
+    across all wires.
+
+    :param network_ansatz: A ``NetworkAnsatz`` class specifying the quantum network simulation.
+    :type network_ansatz: NetworkAnsatz
+
+    :returns: A qnode called as ``qnode(settings)`` for evaluating the density matrix of the
+              network ansatz.
+    :rtype: ``pennylane.QNode``
+    """
+
+    @qml.qnode(qml.device(**network_ansatz.dev_kwargs), **qnode_kwargs)
+    def circuit(settings):
+        network_ansatz.fn(settings)
+        return qml.density_matrix(wires=network_ansatz.network_wires)
```

Comment:
```suggestion
        return qml.density_matrix(wires=network_ansatz.layers_wires[-1])
```

In general, there may be ancillary wires in the network that aren't measured. To be  more consistent with other qnodes use `network_ansatz.layers_wires[-1]`. The current convention in qnetvo is for all measurement nodes to be placed in the final layer, so this will get the density matrix on all measurement wires.